File: community/cdo/src/libcdi/interfaces/python/testLib.py
import CdiLib
ifile = "../testdata/mulval.nc"
streamID = CdiLib.streamOpenRead(ifile)
vlistID  = CdiLib.streamInqVlist(streamID)
nvars    = CdiLib.vlistNvars(vlistID)
for i in range(0,nvars):
  print CdiLib.vlistInqVarCode(vlistID, i),

CdiLib.streamClose(streamID)

File: community/cdo/src/libcdi/interfaces/ruby/testLib.rb
require 'CdiLib'
include CdiLib
ifile = ARGV[0].nil? ? "../testdata/mulval.nc" : ARGV[0]
streamID = streamOpenRead(ifile)
vlistID  = streamInqVlist(streamID)
nvars    = vlistNvars(vlistID)
(0...nvars).each {|i|
  print vlistInqVarCode(vlistID, i).to_s + ' '
}
puts
streamClose(streamID)
